src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
   [k: string]: any;
 }
 
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties?: FullSwapProperties;
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties?: PartialSwapProperties;
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
```

Comment:
Bitcoin's "magic bytes" can be interpreted in a u64 so they could also be a number if we want. Alternatively, we could use the name of the Ethereum network (Ropsten, etc).

This sort of brings back the discussion of how to identify networks. We are not using magic-bytes for Bitcoin at the moment for example ...

Reply:
I actually decided to remove it, the network is not something dynamic really.